app/screens/channel/channel.js

```diff
@@ -115,6 +116,12 @@ export default class Channel extends PureComponent {
             this.loadChannels(nextProps.currentTeamId);
         }
 
+        if (nextProps.currentChannelId !== this.props.currentChannelId &&
+            nextProps.currentTeamId === this.props.currentTeamId &&
```

Comment:
Why check if the team ID didn't change? Shouldn't it check when either the channel or team changes?

Reply:
the channel and the team are changed in different actions, so the team changes first and then the channel I wanted this to get executed only once

Reply:
Oh, that's weird. There's a good chance that that will cause issues because I'd always expect the current channel to be on the current team. I'm going to make a ticket to look into that later